draft-schinazi-quic-version-negotiation.md

```diff
@@ -112,42 +115,143 @@ versions as long as they are reasonably popular.  This is not generally an issue
+  this transport parameter.
+
+Received Negotiation Version Count:
+
+: A variable-length integer specifying the number of Received Negotiation
+  Version fields following it. If the client is sending this Initial in
+  response to a Version Negotiation packet, the subsequent versions SHALL
+  include all the versions from that Version Negotiation packet in order, even
+  if they are not supported by the client (even if the versions are reserved).
+  If the client has not received a Version Negotiation packet on this
+  connection, this field SHALL be 0.
+
+Compatible Version Count:
+
+: A variable-length integer specifying the number of Compatible Version fields
+  following it. The client lists all versions compatible with Attempted Version
+  in the subsequent Compatible Version fields, ordered by descending preference.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                    Negotiated Version (32)                    |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 Supported Version Count (i)                 ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                    Supported Version 1 (32)                   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                   [Supported Version 2 (32)]                  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                               ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                   [Supported Version N (32)]                  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+{: #fig-server-tp title="Server Transport Parameter Format"}
+
+The content of each field is described below:
+
+Negotiated Version:
+
+: The version that the server chose to use for the connection. This field SHALL
+  be equal to the value of the Version field in the long header that carries
+  this transport parameter.
+
+Supported Version Count:
+
+: A variable-length integer specifying the number of Supported Version fields
+  following it. The server encodes all versions it supports in the subsequent
+  list, ordered by descending preference.
+
+
+Clients MAY include versions following the pattern `0x?a?a?a?a` in their
+Compatible Version list, and the server in their Supported Version list.
+Those versions are reserved to exercise version negotiation (see the Versions
+section of {{QUIC}}), and MUST be ignored when parsing these fields. Clients
+MUST NOT add reserved versions that it did not oberve in the Version
+Negotiation packet to the Received Negotiation Version list, and clients MUST
+include any reserved versions seen in the Version Negotiation packet to the
+Received Negotiation Version list.
```

Comment:
Would it be simpler to just say that it must be identical?

Reply:
Fixed